nodejs/private/user_build_settings.bzl

```diff
@@ -0,0 +1,33 @@
+"""Function for user args to be set on the command line
+"""
+
+load("//nodejs/private/providers:user_build_settings.bzl", "UserBuildSettingInfo")
+
+# this can be used to limit user input for flags and also provide useful error messages for incorrect flags
+user_args_allowed = ["--preserve-symlinks", ""]
```

Comment:
I think we should just open this up and let users can set whatever default they want. Guessing they can set a defaults in their .bazelrc file with this improved approach?